bin/cheritest/cheritest_registers.c

```diff
@@ -66,12 +66,17 @@ void
 test_copyregs(const struct cheri_test *ctp __unused)
 {
 
-	CHERI_CGETDEFAULT(2);
-	CHERI_CGETDEFAULT(3);
-	CHERI_CGETDEFAULT(4);
-	CHERI_CGETDEFAULT(5);
-	CHERI_CGETDEFAULT(6);
-	CHERI_CGETDEFAULT(7);
+	__asm__ __volatile__(
+	    ".set push\n"
+	    ".set noreorder\n"
+	    "cgetdefault $c2\n"
+	    "cmove $c3, $c2\n"
+	    "cmove $c4, $c2\n"
+	    "cmove $c5, $c2\n"
+	    "cmove $c6, $c2\n"
+	    "cmove $c7, $c2\n"
+	    ".set pop\n"
+	    : : "i" (2), "i" (3), "i" (4), "i" (5), "i" (6), "i" (7));
```

Comment:
These operands don't seem to be needed? Also I guess we should add explicit clobbers for the registers that are written to.

Reply:
I believe they are clobbers because they are in the 3rd portion.  My understanding is you can't clobber "$c2" directly but instead have to use '"i" (2)'.

Reply:
No, third position is inputs, fourth position is clobbers. All this does is make `%0` through `%6` be immediate values `2` through `7`, as `i` is just a constraint for an immediate argument. It's not currently marking anything as clobbered.

Reply:
And yes, you've been able to clobber `$cN` for a while now. Back before the compiler was CHERI-aware this was the only way you could control which register you were using (`"$c%0" :: i(N)` would be `"$cN"`), but as soon as the compiler becomes CHERI-aware it needs to know about the clobbers.

Reply:
Bah, I was thrown off by the old macros I was removing which (ab)used the immediates to build register names, e.g. '$c%1' paired with '"i" (cb)'.